### PR TITLE
fix hostname and hostnameFQDN filters

### PR DESCRIPTION
**Pull request by stuartnelson3**
hostname filter was broken if a scheme was not present.

@grobie

### CODE REVIEW

File: app/assets/javascripts/angular/filters/filters.js [hunk 1]
```diff
@@ -12,17 +12,23 @@ angular.module("Prometheus.filters").filter('toPercentile', function() {
...
 angular.module("Prometheus.filters").filter('hostname', function() {
   return function(input) {
-    var a = document.createElement("a");
-    a.href = input;
-    return a.host.split(".", 1)[0];
+    var re = /^([^:\/?]+:\/\/(www\.)?)?([^\/]+)/;
+    var match = input.match(re);
+    if (match && match[3]) {
+      return match[3].replace('www.', '').replace(/:[0-9]+/,'').split('.')[0];
+    }
```
Comment (grobie) [inline app/assets/javascripts/angular/filters/filters.js]: Could we just call the hostname function here, instead of duplicating the code?

pseudo code:

```
$filter('hostnameFqdn')(input).split('.')[0]
```

Comment (stuartnelson3) [inline app/assets/javascripts/angular/filters/filters.js]: yeah, probably

File: spec/javascripts/angular/filters/filters_spec.js [hunk 2]
```diff
@@ -16,12 +16,32 @@ describe('filters', function() {
...
+  describe('hostname', function() {
+    it('with protocol', function() {
+      var host = $filter('hostname');
+      [
+        'http://hostname.domain.com:9090/path?search',
+        'http://hostname.domain.com/path?search',
+        'ws://hostname.domain:9000/path?search',
+        'hostname.domain.com:9090/path?search',
+        'hostname.domain.com/path?search',
+        'hostname.domain:9000/path?search'
```
Comment (grobie) [inline spec/javascripts/angular/filters/filters_spec.js]: Let's add `hostname.domain:9000` and `hostname.domain`. And `hostname` :)

File: spec/javascripts/angular/filters/filters_spec.js [hunk 3]
```diff
@@ -16,12 +16,32 @@ describe('filters', function() {
...
+      ].forEach(function(url) {
+        expect(host(url)).toEqual('sub-domain.domain.com');
+      });
   });
 
-  it('host', function() {
-    var host = $filter('hostname');
-    expect(host('http://domain.com/path?search')).toEqual('domain');
+  describe('hostname', function() {
+    it('with protocol', function() {
```
Comment (grobie) [inline spec/javascripts/angular/filters/filters_spec.js]: That's not correct anymore, no? Maybe just revert it to `it('hostname', ...` ?

File: spec/javascripts/angular/filters/filters_spec.js [hunk 4]
```diff
@@ -31,12 +51,13 @@ describe('filters', function() {
     });
 
     it('exact match', function() {
-      expect(regex('important-data.generic-data', '\.generic-data', '')).toEqual('important-data');
+      expect(regex('important-data.generic-data', '.generic-data', '')).toEqual('important-data');
```
Comment (grobie) [inline spec/javascripts/angular/filters/filters_spec.js]: Why are these changes part of this commit? Doesn't seem to be related to hostname changes.

**Comment by grobie** [comment]
:+1: pending the comments and removal of non-hostname related code changes.

File: spec/javascripts/angular/filters/filters_spec.js (hunk 4 above)
Comment (stuartnelson3) [inline spec/javascripts/angular/filters/filters_spec.js]: it's not, it was just cleanup in the spec file

**Comment by stuartnelson3** [comment]
merged without non-hostname code changes
